pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/BaseRecordExtractor.java

```diff
@@ -186,6 +187,9 @@ protected Object convertSingleValue(Object value) {
     if (value instanceof Number || value instanceof byte[]) {
       return value;
     }
+    if (value instanceof GenericData.Fixed) {
```

Comment:
Nit: Why not check for the interface _GenericFixed_ instead? I think the same logic would also apply for other implementations of GenericFixed, e.g. [SpecificFixed](https://avro.apache.org/docs/1.8.1/api/java/org/apache/avro/specific/SpecificFixed.html).

Reply:
Done.